### ISSUE TITLE
Bug validation of array

### ISSUE THREAD

**Issue by silverpark** [Respect/Validation#425]
Hi,

There is an error of exception message of validation.

For this simple code exemple :
```php
$input = array(
  "age" => 1,
  "reference" =>  "QSF1234",
);

try {
        v::create()
            ->key(
                'age',
                v::int()->notEmpty()->noneOf(v::string()),
                true
            )
            ->key(
                'reference',
                v::string()->notEmpty()->length(1, 50, true),
                true
            )
            ->assert($input);
            ;
} catch (NestedValidationExceptionInterface $e) {
    var_dump( $e->getFullMessage() );
}
```

If I test this array :
```php
$input = array(
);
```

I receive this error
```
string(117) "\-These rules must pass for "Array"
  |-Key age must be present
  \-Key reference must be present"
```

It's ok, But If I missing a field like this array :
```php
$input = array (
  "age" => 1,
);
```

OR

```php
$input = array (
  "reference" =>  "QSF1234",
);
```

I receive an error without specification :
```
string(35) "\-These rules must pass for "Array""
```

I think that, this bug is present when only one field is missing, no more.

I would like receive an error with specification like ``Key age must be present`` or ``Key reference must be present``

**Comment by henriquemoody** [comment]
This is duplicated of #179.

Any way, if you use `check()` instead of `assert()` you're gonna get this message but not wall the failures will be shown.

Personally I use `check()` for most cases, because `assert()` doesn't display a very user friendly message, IMO.

**Comment by silverpark** [comment]
I'm sorry but too many things are not good about of return error exception.

Can you give me all possibilities of variables exists in the exception message ? like exemple ``{{name}}``

**Comment by henriquemoody** [comment]
The possibilities depends on each validator, you must check the exception class of each of them you're using.

There are many things to fix yet until the next release and dev-master is much better than latest versions but I do not recommend you use it for a while, there will be more BC breaks util 24th.

**Comment by silverpark** [comment]
thank you, yet I must to use dev-master branch, your new version is best !

**Comment by silverpark** [comment]
This error about this issue is fixed in master branch with the assert function but why when I field is missing the error is not show but only when the value is missing ?

**Comment by henriquemoody** [comment]
I have to check it out, unfortunately I think I'm just going to be able to do that next week.

**Comment by silverpark** [comment]
Ok, I will be notify about commits, thank you very much for all your support/debug etc... 

Great :+1:

**Comment by henriquemoody** [comment]
You're welcome, @silverpark.

:panda_face:

**Comment by silverpark** [comment]
Hi, 

What's news ?

**Comment by henriquemoody** [comment]
No news yet, but I will not release 1.0 until I fix it, specially because it required a big refactoring on exceptions.

**Comment by silverpark** [comment]
ok, thanks

**Comment by henriquemoody** [comment]
This fix will be released on Saturday, Oct 24th, with version 1.0:

This test was created to ensure this is working: https://github.com/Respect/Validation/blob/master/tests/integration/issue-425.phpt

**Comment by silverpark** [comment]
Great ! Thank you very much !
